internal/buildengine/engine.go

```diff
@@ -1352,6 +1353,7 @@ func (e *Engine) newModuleMeta(ctx context.Context, config moduleconfig.Unvalida
 		plugin:         plugin,
 		events:         events,
 		configDefaults: customDefaults,
+		watch:          watch.NewWatcher(optional.Zero(validConfig.BuildLock), validConfig.Watch...),
```

Comment:
Is there a race condition using the build lock here?
Using the build lock here could also cause race conditions. You want the watcher to receive the modified file hashes before it starts polling for changes again. But the build lock is unlocked when the plugin sends the build finished message... but that doesnt mean the engine has handled the build result yet.